proc-macro/src/lib.rs

```diff
@@ -751,7 +762,32 @@ pub mod token_stream {
         type IntoIter = IntoIter;
 
         fn into_iter(self) -> Self::IntoIter {
-            IntoIter(self.inner.into_iter())
+            IntoIter {
+                range: 0..self.inner.len(),
+                stream: self,
+            }
+        }
+    }
+}
+
+// Look up `s` in our global table of interned strings. If it exists return that
+// string, otherwise allocate `s` onto the heap, leak it, and then return it.
+fn intern<T>(s: T) -> &'static str
```

Comment:
With the instance-caching logic from #21, memory leaked by the wasm module now lives for the full lifetime of the thread it was run on, instead of a single invocation. That's probably fine (rustc is already interning all of these), but I figure is worth noting.

Reply:
True yeah! We may want to eventually add hooks for proc macros to expose to "shut down" and free internal state, such as freeing wasm modules, if it becomes an issue. I think for the duration of the macro expansion process though it's fine to basically leak all this information.